plugins/BEdita/Core/src/Cache/LayeredEngine.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2021 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+namespace BEdita\Core;
+
+use Cake\Cache\Cache;
+use Cake\Cache\CacheEngine;
+use Cake\Cache\Engine\ArrayEngine;
+use Exception;
+
+/**
+ * This engine uses two layers of cache, one persistent and one in-memory for faster lookup times.
+ */
+class LayeredEngine extends CacheEngine
+{
+    /**
+     * Persistent cache instance.
+     *
+     * @var CacheEngine
+     */
+    protected $_persistent = null;
+
+    /**
+     * In-memory cache instance.
+     *
+     * @var CacheEngine
```

Comment:
[minor] We can be more specific here. Also, in docblock annotations I usually prefer FQCN rather than aliases, which are more concise but are dependent on current namespace and file `use`d classes.

```suggestion
     * @var \Cake\Cache\Engine\ArrayEngine
```